front/src/components/AdminPanel/Panel/CompetitionDetails/CompetitionDetails.js

```diff
@@ -4,13 +4,21 @@ import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 import competitionDetaildata from "../../competitionsData.json";
+import { revertDateFormat } from "../../../revertDateFormat.js";
 
 function CompetitionDetails(props) {
     const _id = props.match.params.id;
-    const name = competitionDetaildata[_id - 1]["name"];
-    const location = competitionDetaildata[_id - 1]["location"];
-    const start_date = competitionDetaildata[_id - 1]["date_start"];
-    const end_date = competitionDetaildata[_id - 1]["date_end"];
+
+    let desiredCompetition = {};
+    Object.keys(competitionDetaildata).forEach((oneComp) => {
+        if (competitionDetaildata[oneComp].id === parseInt(_id))
+            desiredCompetition = competitionDetaildata[oneComp];
+    });
```

Comment:
Isn't it just over-complicated version of:
```suggestion
    const desiredCompetition = competitionDetaildata.find(comp => comp.id === parseInt(_id));
```
?